### ISSUE TITLE
Address already in use (now with international flavor!)

### ISSUE THREAD

**Issue by ichorid** [Tribler/tribler#5437]
7.5.1:

```python
Traceback (most recent call last):
File "src\tribler-gui\tribler_gui\event_request_manager.py", line 112, in on_read_data
RuntimeError: [Errno 10048] error while attempting to bind on address ('127.0.0.1', 45777): normalerweise darf jede socketadresse (protokoll, netzwerkadresse oder anschluss) nur jeweils einmal verwendet werden
--LONG TEXT--
Traceback (most recent call last):
File "lib\asyncio\events.py", line 88, in _run
File "ipv8\taskmanager.py", line 123, in done_cb
File "src\tribler-core\tribler_core\modules\tunnel\socks5\server.py", line 27, in start
File "lib\asyncio\base_events.py", line 1374, in create_server
OSError: [Errno 10048] error while attempting to bind on address ('127.0.0.1', 45777): normalerweise darf jede socketadresse (protokoll, netzwerkadresse oder anschluss) nur jeweils einmal verwendet werden

--CONTEXT--
{'message': 'Exception in callback TaskManager.register_task.<locals>.done_cb(<Task finishe...ndet werden")>) at ipv8\\taskmanager.py:120', 'exception': OSError(10048, "error while attempting to bind on address ('127.0.0.1', 45777): normalerweise darf jede socketadresse (protokoll, netzwerkadresse oder anschluss) nur jeweils einmal verwendet werden"), 'handle': <Handle TaskManager.register_task.<locals>.done_cb(<Task finishe...ndet werden")>) at ipv8\taskmanager.py:120>}
```

```python
Traceback (most recent call last):
File "src\tribler-gui\tribler_gui\event_request_manager.py", line 112, in on_read_data
RuntimeError: [Errno 10048] error while attempting to bind on address ('127.0.0.1', 9150): обычно разрешается только одно использование адреса сокета (протокол/сетевой адрес/порт)
--LONG TEXT--
Traceback (most recent call last):
File "lib\asyncio\events.py", line 88, in _run
File "ipv8\taskmanager.py", line 123, in done_cb
File "src\tribler-core\tribler_core\modules\tunnel\socks5\server.py", line 27, in start
File "lib\asyncio\base_events.py", line 1374, in create_server
OSError: [Errno 10048] error while attempting to bind on address ('127.0.0.1', 9150): обычно разрешается только одно использование адреса сокета (протокол/сетевой адрес/порт)

--CONTEXT--
{'message': 'Exception in callback TaskManager.register_task.<locals>.done_cb(<Task finishe...адрес/порт)")>) at ipv8\\taskmanager.py:120', 'exception': OSError(10048, "error while attempting to bind on address ('127.0.0.1', 9150): обычно разрешается только одно использование адреса сокета (протокол/сетевой адрес/порт)"), 'handle': <Handle TaskManager.register_task.<locals>.done_cb(<Task finishe...адрес/порт)")>) at ipv8\taskmanager.py:120>}
```

```


Traceback (most recent call last):
File "src\tribler-gui\tribler_gui\event_request_manager.py", line 112, in on_read_data
RuntimeError: [Errno 10048] error while attempting to bind on address ('127.0.0.1', 9983): 각 소켓 주소(프로토콜/네트워크 주소/포트)는 하나만 사용할 수 있습니다
--LONG TEXT--
Traceback (most recent call last):
File "lib\asyncio\events.py", line 88, in _run
File "ipv8\taskmanager.py", line 123, in done_cb
File "src\tribler-core\tribler_core\modules\tunnel\socks5\server.py", line 27, in start
File "lib\asyncio\base_events.py", line 1374, in create_server
OSError: [Errno 10048] error while attempting to bind on address ('127.0.0.1', 9983): 각 소켓 주소(프로토콜/네트워크 주소/포트)는 하나만 사용할 수 있습니다

--CONTEXT--
{'message': 'Exception in callback TaskManager.register_task.<locals>.done_cb(<Task finishe... 사용할 수 있습니다")>) at ipv8\\taskmanager.py:120', 'exception': OSError(10048, "error while attempting to bind on address ('127.0.0.1', 9983): 각 소켓 주소(프로토콜/네트워크 주소/포트)는 하나만 사용할 수 있습니다"), 'handle': <Handle TaskManager.register_task.<locals>.done_cb(<Task finishe... 사용할 수 있습니다")>) at ipv8\taskmanager.py:120>}
```

And some Jenkins stuff:
```python
Traceback (most recent call last):
File "/home/tribler/jenkins/workspace/Build-Tribler_release/Async-Py3-Build-Ubuntu64/tribler/src/tribler-gui/tribler_gui/core_manager.py", line 119, in on_received_state
RuntimeError: [Errno 98] Address already in use
--LONG TEXT--
Traceback (most recent call last):
File "/home/tribler/jenkins/workspace/Build-Tribler_release/Async-Py3-Build-Ubuntu64/tribler/src/run_tribler.py", line 95, in start_tribler
File "/home/tribler/jenkins/workspace/Build-Tribler_release/Async-Py3-Build-Ubuntu64/tribler/src/tribler-core/tribler_core/session.py", line 396, in start
File "/home/tribler/jenkins/workspace/Build-Tribler_release/Async-Py3-Build-Ubuntu64/tribler/src/tribler-core/tribler_core/restapi/rest_manager.py", line 75, in start
File "/home/tribler/.local/lib/python3.6/site-packages/aiohttp/web_runner.py", line 104, in start
File "/usr/lib/python3.6/asyncio/base_events.py", line 1082, in create_server
sock.listen(backlog)
OSError: [Errno 98] Address already in use

--CONTEXT--
{'message': 'Task exception was never retrieved', 'exception': OSError(98, 'Address already in use'), 'future': <Task finished coro=<start_tribler_core.<locals>.start_tribler() done, defined at /home/tribler/jenkins/workspace/Build-Tribler_release/Async-Py3-Build-Ubuntu64/tribler/src/run_tribler.py:64> exception=OSError(98, 'Address already in use')>
```

**Comment by devos50** [comment]
All of them, except for the last one, seem to be related to the SOCKS5 server. IIRC, we first verify is a port is actually available. But it might be that that port is claimed by another process/module later.

As an alternative, the `start` method in `server.py` could return a list of ports that it has claimed, and retry if a port is already occupied.

The last error seems to indicate that the default (HTTP) port 8085 is already occupied on the users' machine. I'm not sure how much there is we can do to resolve that.

**Comment by ichorid** [comment]
The proper fix for this will require changing the logic of claiming the ports to be atomic and move it to the SOCKS servers. This is indeed possible, but too invasive for 7.5.

**Comment by ichorid** [comment]
Solved by SOCKS refactoring in #6206
